File: lingting-core/src/main/java/live/lingting/framework/time/DatePattern.java
package live.lingting.framework.time;

import lombok.experimental.UtilityClass;

import java.time.Instant;
import java.time.ZoneId;
import java.time.ZoneOffset;
import java.time.format.DateTimeFormatter;

/**
 * @author lingting
 */
@UtilityClass
public class DatePattern {

	public static final ZoneOffset DEFAULT_ZONE_OFFSET = ZoneOffset.of("+8");

	public static final ZoneId DEFAULT_ZONE_ID = DEFAULT_ZONE_OFFSET.normalized();

	public static final ZoneOffset GZM_ZONE_OFFSET = ZoneOffset.of("+0");

	public static final ZoneId GZM_ZONE_ID = GZM_ZONE_OFFSET.normalized();

	public static final ZoneId SYSTEM_ZONE_ID = ZoneId.systemDefault();

	public static final ZoneOffset SYSTEM_ZONE_OFFSET = SYSTEM_ZONE_ID.getRules().getOffset(Instant.now());

	public static final String NORM_DATETIME_PATTERN = "yyyy-MM-dd HH:mm:ss";

	public static final String NORM_DATE_PATTERN = "yyyy-MM-dd";

	public static final String NORM_TIME_PATTERN = "HH:mm:ss";

	public static final DateTimeFormatter FORMATTER_YMD_HMS = DateTimeFormatter.ofPattern(NORM_DATETIME_PATTERN);

	public static final DateTimeFormatter FORMATTER_YMD = DateTimeFormatter.ofPattern(NORM_DATE_PATTERN);

	public static final DateTimeFormatter FORMATTER_HMS = DateTimeFormatter.ofPattern(NORM_TIME_PATTERN);

}

File: lingting-core/src/main/kotlin/live/lingting/framework/time/DatePattern.kt
package live.lingting.framework.time

import java.time.Instant
import java.time.ZoneId
import java.time.ZoneOffset
import java.time.format.DateTimeFormatter

/**
 * @author lingting
 */
object DatePattern {

    @JvmField
    val UTC8_ZONE_OFFSET: ZoneOffset = ZoneOffset.of("+8")

    @JvmField
    val UTC8_ZONE_ID: ZoneId = UTC8_ZONE_OFFSET.normalized()

    @JvmField
    val GMT_ZONE_OFFSET: ZoneOffset = ZoneOffset.of("+0")

    @JvmField
    val GMT_ZONE_ID: ZoneId = GMT_ZONE_OFFSET.normalized()

    @JvmField
    val SYSTEM_ZONE_ID: ZoneId = ZoneId.systemDefault()

    @JvmField
    val SYSTEM_ZONE_OFFSET: ZoneOffset = SYSTEM_ZONE_ID.rules.getOffset(Instant.now())

    const val NORM_DATETIME_PATTERN: String = "yyyy-MM-dd HH:mm:ss"

    const val MILLIS_NORM_DATETIME_PATTERN: String = "yyyy-MM-dd HH:mm:ss.SSS"

    const val NORM_DATE_PATTERN: String = "yyyy-MM-dd"

    const val NORM_TIME_PATTERN: String = "HH:mm:ss"

    const val MILLIS_NORM_TIME_PATTERN: String = "HH:mm:ss.SSS"

    const val ISO_8601_DATETIME_PATTERN: String = "yyyy-MM-dd'T'HH:mm:ss'Z'"

    const val MILLIS_ISO_8601_DATETIME_PATTERN: String = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"

    @JvmField
    val FORMATTER_YMD_HMS: DateTimeFormatter = DateTimeFormatter.ofPattern(NORM_DATETIME_PATTERN)

    @JvmField
    val FORMATTER_YMD_HMS_MILLIS: DateTimeFormatter = DateTimeFormatter.ofPattern(MILLIS_NORM_DATETIME_PATTERN)

    @JvmField
    val FORMATTER_YMD: DateTimeFormatter = DateTimeFormatter.ofPattern(NORM_DATE_PATTERN)

    @JvmField
    val FORMATTER_HMS: DateTimeFormatter = DateTimeFormatter.ofPattern(NORM_TIME_PATTERN)

    @JvmField
    val FORMATTER_HMS_MILLIS: DateTimeFormatter = DateTimeFormatter.ofPattern(MILLIS_NORM_TIME_PATTERN)

    @JvmField
    val FORMATTER_ISO_8601: DateTimeFormatter = DateTimeFormatter.ofPattern(ISO_8601_DATETIME_PATTERN)

    @JvmField
    val FORMATTER_ISO_8601_MILLIS: DateTimeFormatter = DateTimeFormatter.ofPattern(MILLIS_ISO_8601_DATETIME_PATTERN)

}

